### PR TITLE
Refactor errors: follow plugin #791

### PR DESCRIPTION
**Pull request by s-medvedev**

### CODE REVIEW

**Comment by s-medvedev** [comment]
Define PLUGIN_CHECK_LOGIC in plugin headers it's bad idea. Main plugin headers included simultaneously and this macros redifined. I work on another method to place "plugin" in logic_exception.

File: tests/plugin_tests/follow.cpp [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+BOOST_AUTO_TEST_CASE(follow_apply) {
+    BOOST_TEST_MESSAGE("Testing: follow_apply");
+
+    ACTORS((alice)(bob));
+
+    generate_blocks(60 / STEEMIT_BLOCK_INTERVAL);
+
+    custom_binary_operation cop;
```
Comment (zxcat) [inline tests/plugin_tests/follow.cpp]: Maybe we should test `custom_json_operation` path too. At least in one plugin, which use custom operations.

File: tests/common/helpers.hpp [hunk 2]
```diff
@@ -29,7 +29,8 @@
 
 // Check operation authorities
 #define CHECK_OP_AUTHS(OP, OWNER, ACTIVE, POSTING) {               \
-    account_name_set owner_auths, active_auths, posting_auths;     \
+    boost::container::flat_set<golos::protocol::account_name_type> \
```
[the remark this answers is not in the document]
Comment (s-medvedev) [reply tests/common/helpers.hpp]: You are right. We have not two... MANY aliases for `flat_set<account_name_type>` :). Fixed.

File: tests/plugin_tests/follow.cpp (hunk 1 above)
Comment (s-medvedev) [reply tests/plugin_tests/follow.cpp]: Ideally, it's must be done in tests for custom_json_operations.

File: tests/common/database_fixture.hpp [hunk 3]
```diff
@@ -421,6 +421,13 @@ namespace golos { namespace chain {
 
         using namespace golos::protocol;
 
+        fc::variant_object make_comment_id(const std::string& author, const std::string& permlink);
+        fc::variant_object make_limit_order_id(const std::string& author, uint32_t orderid);
+        fc::variant_object make_convert_request_id(const std::string& account, uint32_t requestid);
+        fc::variant_object make_escrow_id(const string& name, uint32_t escrow_id);
+
+        typedef fc::flat_set<account_name_type> account_name_set;
```
Comment (afalaleev) [inline tests/common/database_fixture.hpp]: Prefer to use `using account_name_set = fc::flat_set<account_name_type>;`

**Pull request merged**